Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Smoothing/constraints_map.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Konstantinos Katrioplas (konst.katrioplas@gmail.com)
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_CONSTRAINTS_MAP_H
+#define CGAL_POLYGON_MESH_PROCESSING_CONSTRAINTS_MAP_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <CGAL/property_map.h>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace internal {
+
+template<typename Descriptor>
+struct Constrained_vertices_map
+{
+  typedef Descriptor                          key_type;
+  typedef bool                                value_type;
+  typedef value_type&                         reference;
+  typedef boost::read_write_property_map_tag  category;
+
+  boost::shared_ptr<std::set<Descriptor> > constrained_data;
+
+public:
+  Constrained_vertices_map() : constrained_data(new std::set<Descriptor>) {}
+
+  friend bool get(const Constrained_vertices_map& map, const key_type& d)
+  {
+    typename std::set<Descriptor>::iterator it = map.constrained_data->find(d);
+    return it != map.constrained_data->end() ? true : false;
+  }
+
+  friend void put(Constrained_vertices_map& map, const key_type& d)
```

Comment:
I doubt this will work, you are missing the value type. Anyway this is not needed as you only want a read-only map + it is a duplicate of `CGAL::Boolean_property_map` so the whole class can be removed.